src/python_testing/TC_SWTCH.py

```diff
@@ -361,6 +372,90 @@ def _received_event(self, event_listener: EventChangeCallback, target_event: Clu
+                TestStep(6, "TH reads the CurrentPosition attribute from the DUT", "Verify that the value is 1"),
+                TestStep(7, "If there are more than 2 positions, test subsequent positions of the DUT"),
+                TestStep(8, "Operator sets switch to first position on the DUT."),
+                TestStep(9, "Wait 10 seconds for event reports stable." "Verify that last SwitchLatched event received is for NewPosition 0."),
+                TestStep(10, "TH reads the CurrentPosition attribute from the DUT", "Verify that the value is 0"),
+                ]
+
+    @per_endpoint_test(has_feature(Clusters.Switch, Clusters.Switch.Bitmaps.Feature.kLatchingSwitch))
+    async def test_TC_SWTCH_2_2(self):
+        post_prompt_settle_delay_seconds = 10.0
+
+        # Step 1: Commissioning - already done
+        self.step(1)
+
+        cluster = Clusters.Switch
+        endpoint_id = self.matter_test_config.endpoint
+
+        # Step 2: Set up subscription to all events of Switch cluster on the endpoint.
+        self.step(2)
+        event_listener = EventChangeCallback(cluster)
+        await event_listener.start(self.default_controller, self.dut_node_id, endpoint=endpoint_id)
+
+        # Step 3: Operator sets switch to first position on the DUT.
+        self.step(3)
+        self._ask_for_switch_position(endpoint_id, new_position=0)
+        event_listener.flush_events()
+
+        # Step 4: TH reads the CurrentPosition attribute from the DUT.
+        # Verify that the value is 0.
+        self.step(4)
+        button_val = await self.read_single_attribute_check_success(cluster=cluster, attribute=cluster.Attributes.CurrentPosition)
+        asserts.assert_equal(button_val, 0, "Switch position value is not 0")
+
+        # Step 5: Operator sets switch to second position (one) on the DUT",
+        # Verify that the TH receives SwitchLatched event with NewPosition set to 1 from the DUT
+        self.step(5)
+        expected_switch_position = 1
+        self._ask_for_switch_position(endpoint_id, expected_switch_position)
+
+        data = event_listener.wait_for_event_report(cluster.Events.SwitchLatched, timeout_sec=post_prompt_settle_delay_seconds)
+        logging.info(f"-> SwitchLatched event last received: {data}")
+        asserts.assert_equal(data, cluster.Events.SwitchLatched(
+            newPosition=expected_switch_position), "Did not get expected switch position")
+
+        # Step 6: TH reads the CurrentPosition attribute from the DUT", "Verify that the value is 1
+        self.step(6)
+        button_val = await self.read_single_attribute_check_success(cluster=cluster, attribute=cluster.Attributes.CurrentPosition)
+        asserts.assert_equal(button_val, expected_switch_position, f"Switch position is not {expected_switch_position}")
+
+        # Step 7: If there are more than 2 positions, test subsequent positions of the DUT
+        # TODO: Implement loop for > 2 total positions
```

Comment:
Can you link a follow up for this? just worried it'll get missed in the post te rush.

Reply:
Added follow-up